lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -81,6 +81,9 @@ defmodule Mix.Compilers.Elixir do
 
     {force?, stale, new_deps_config} =
       cond do
+        System.get_env("MIX_NO_COMPILE") in ["1", "true"] ->
```

Comment:
We should probably place this inside compile.all. This would be specific to Elixir and other compilers would still run, and i think none should.

Reply:
Makes sense, will fix!